llvm/include/llvm/Transforms/Utils/SampleProfileLoaderBaseImpl.h

```diff
@@ -464,17 +461,8 @@ SampleProfileLoaderBaseImpl<BT>::getProbeWeight(const InstructionT &Inst) {
     return std::error_code();
 
   const FunctionSamples *FS = findFunctionSamples(Inst);
-  // If none of the instruction has FunctionSample, we choose to return zero
-  // value sample to indicate the BB is cold. This could happen when the
-  // instruction is from inlinee and no profile data is found.
-  // FIXME: This should not be affected by the source drift issue as 1) if the
-  // newly added function is top-level inliner, it won't match the CFG checksum
-  // in the function profile or 2) if it's the inlinee, the inlinee should have
-  // a profile, otherwise it wouldn't be inlined. For non-probe based profile,
-  // we can improve it by adding a switch for profile-sample-block-accurate for
-  // block level counts in the future.
   if (!FS)
-    return 0;
+    return std::error_code();
```

Comment:
Can we keep a simple comment here? something like this: not having a sample indicates ... and we treat it as unknown, leaving it to profile inference instead of forcing a zero count.